doc/source/optimization.md

```diff
@@ -1,67 +1,264 @@
+This means that if you have three user placeholders running, real users will
+only need to wait for a scale up if more than three users arrive in an interval
+of time less than it takes to make a node ready for use.
+
+To use three user placeholders for example, that can do their job thanks to pod
+priority, add the following configuration:
+
+
+```yaml
+scheduling:
+  podPriority:
+    enabled: true
+  userPlaceholder:
+    replicas: 3
+```
+
+For further discussion about user placeholders, see [@MinRK's excellent
+post](https://discourse.jupyter.org/t/planning-placeholders-with-jupyterhub-helm-chart-0-8-tested-on-mybinder-org/213)
+about it where he analyzed its introduction on mybinder.org.
+
+IMPORTANT: Further settings may be required for successful use of the pod priority depending on how your cluster autoscaler is configured. This is know to work on GKE, but we don't know how it works on other cloud providers or kubernetes installation. See the [configuration reference](reference.html#scheduling-podpriority) for more details.
+
+### Scaling down efficiently
+
+Scaling up is the easy part, scaling down is harder. To scale down a node,
+[certain technical
+criteria](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/FAQ.md#what-types-of-pods-can-prevent-ca-from-removing-a-node)
+needs to be met. The central one is that the node to be scaled down must be free
+of pods that isn't allowed to be disrupted. Pods that are not allowed to be
+disrupted are for example real user pods, various system pods, and some
+JupyterHub pods (without a permissive
+[PodDisruptionBudget](https://kubernetes.io/docs/concepts/workloads/pods/disruptions/)).
+Consider for example that you add a lot of users during the daytime. New nodes
+are added by the CA. Some system pod ends up on the new nodes along with the
+user pods for some reason. At night when the
+[*culler*](user-management.html#culling-user-pods) has removed many inactive
+pods by some nodes was free from users but there was still a single system pod
+stopping the CA from removing it.
+
+To avoid these scale down failures, we recommend using a *dedicated node pool*
+for the user pods. To accomplish this, we can use [*taints and
+tolerations*](https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/).
+If we add a taint to all the nodes in the node pool, and a toleration on the
+user pods to tolerate being scheduled on a tainted node, we have practically
+dedicated the node pool to be used by user pods.
+
+#### Using a user dedicated node pool
+
+To make users schedule on a dedicated node for them, you need to do the following:
+
+1. Setup a node pool (with autoscaling), a certain label, and a certain taint.
+
+    How you do this will depend on the cloud provider, also note that cloud
+    providers often have their own labels separate from kubernetes labels. This
+    needs to be a kubernetes label.
+
+    ```
```

Comment:
Filename?

Reply:
Ah yeah no file, ill inline these I guess as this indicates a file is to be written.

Reply:
Ah yeah no file, ill inline these I guess as this indicates a file is to be written.

Reply:
Ah yeah no file, ill inline these I guess as this indicates a file is to be written.